unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarLOD/AvatarLODController/AvatarLODController.cs

```diff
@@ -154,6 +155,17 @@ private void SetAvatarFeatures(bool newSSAOEnabled, bool newFacialFeaturesEnable
             }
         }
 
+        public void UpdateImpostorTint(float distanceToMainPlayer, float minImpostorTintDistance, float maxImpostorTintDistance, float nearestImpostorColorTint, float farestImpostorColorTint, float nearestImpostorAlphaValue, float farestImpostorAlphaValue)
+        {
+            float initialStep = Mathf.Max(minImpostorTintDistance, distanceToMainPlayer);
+            float tintStep = Mathf.InverseLerp(minImpostorTintDistance, maxImpostorTintDistance, initialStep);
+            float tintValue = Mathf.Lerp(nearestImpostorColorTint, farestImpostorColorTint, tintStep);
+            Color newColor = Color.Lerp(Color.white, Color.black, tintValue);
+            newColor.a = Mathf.Lerp(nearestImpostorAlphaValue, farestImpostorAlphaValue, tintStep);
+
+            player.renderer.SetImpostorColor(newColor);
+        }
+
```

Comment:
I would do a couple of things with this. 

I love the use of `DataStore` but only when the data really needs to be accessed accross systems. In this case, besides the development tool panel, we won't ever change this at runtime.

So, first I would move all the variables for the impostor tint from the `DataStore` to _constants_ in another class (probably `AvatarRendererHelpers`).

Then I would use a Helper (also in AvatarRendererHelpers) function to avoid having all this boilerplate in this class:
```
public Color CalculateImpostorTint(float distanceToMainPlayer, 
    float minImpostorTintDistance = MIN_IMPOSTOR_TINT_DISTANCE, 
    float maxImpostorTintDistance = MAX_IMPOSTOR_TINT_DISTANCE, 
    float nearestImpostorColorTint = NEAREST_IMPOSTOR_TINT_COLOR, 
    float farestImpostorColorTint =  FARTHEST_IMPOSTOR_COLOR_TINT,
    float nearestImpostorAlphaValue = NEAREST_IMPOSTOR_ALPHA_VALUE,
    float farestImpostorAlphaValue = FARTHEST_IMPOSTOR_ALPHA_VALUE)
{
    float initialStep = Mathf.Max(minImpostorTintDistance, distanceToMainPlayer);
    float tintStep = Mathf.InverseLerp(minImpostorTintDistance, maxImpostorTintDistance, initialStep);
    float tintValue = Mathf.Lerp(nearestImpostorColorTint, farestImpostorColorTint, tintStep);
    Color newColor = Color.Lerp(Color.white, Color.black, tintValue);
    newColor.a = Mathf.Lerp(nearestImpostorAlphaValue, farestImpostorAlphaValue, tintStep);
    return newColor;
}
```
So you should only have to call `AvatarRendererHelpers.CalculateImpostorTint(distanceToPlayer);`.
The only reason I'm adding these params are because maybe it makes easier for you to test this function but if the default values are valid for testing, I would remove the param and use the constants directly inside the method.

Reply:
I added the default values to the `AvatarRendererHelpers` class and use them directly inside `UpdateImpostorTint()`, since we don't really need those values as parameters